src/widget/wwaveformviewer.cpp

```diff
@@ -208,13 +246,38 @@ void WWaveformViewer::setPlayMarkerPosition(double position) {
 void WWaveformViewer::setWaveformWidget(WaveformWidgetAbstract* waveformWidget) {
     if (m_waveformWidget) {
         QWidget* pWidget = m_waveformWidget->getWidget();
-        disconnect(pWidget, SIGNAL(destroyed()),
-                   this, SLOT(slotWidgetDead()));
+        disconnect(pWidget, &QWidget::destroyed, this, &WWaveformViewer::slotWidgetDead);
     }
     m_waveformWidget = waveformWidget;
     if (m_waveformWidget) {
         QWidget* pWidget = m_waveformWidget->getWidget();
-        connect(pWidget, SIGNAL(destroyed()),
-                this, SLOT(slotWidgetDead()));
+        connect(pWidget, &QWidget::destroyed, this, &WWaveformViewer::slotWidgetDead);
+        m_waveformWidget->getWidget()->setMouseTracking(true);
     }
 }
+
+CuePointer WWaveformViewer::getCuePointerFromCueMark(WaveformMarkPointer pMark) {
+    if (pMark) {
+        QList<CuePointer> cueList = m_waveformWidget->getTrackInfo()->getCuePoints();
```

Comment:
I suggest to add the following debug assertion:

```
DEBUG_ASSERT(pMark->getHotCue() != Cue::kNoHotCue);
```

Just to ensure that the mark always has a number that uniquely identifies a `Cue` object.

Reply:
This will fail currently. I can right click on loop markers with no hotcue number, which should not happen.